### PR TITLE
Review/modules exercise

### PR DESCRIPTION
**Pull request by chfsx**
Hello, 

I completed the review of the `Modules/Exercise` component.

Results:

- [x] The code style is `PSR-2 + X` compliant
- [ ] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`: There are 3 usages of POST
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant): not relevant
- [ ] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties): There are ~150 missing or wrong declarations in ~38 files (see PR)

--

Actions needed:
- [ ] Please check my inline comments. Some of them are just information, some require an action. You can use `TODO PHP8-REVIEW` when searching. The found super globals might be okay if there is a **good** reason to keep them.
- [ ] in commit 11c7efca1ed755a21476116c122af6e3c5c9686c there are several changes such as sizeof->count or others. feel free to truncate or pick them

Best regards,
@chfsx

### CODE REVIEW

**Comment by alex40724** [comment]
@chfsx  Thanks for the review. One thing I just saw. The license header you are using are not according to the JF decision.

JF:
```
/**
 * This file is part of ILIAS, a powerful learning management system
 * published by ILIAS open source e-Learning e.V.
 *
 * ILIAS is licensed with the GPL-3.0,
 * see https://www.gnu.org/licenses/gpl-3.0.en.html
 * You should have received a copy of said license along with the
 * source code, too.
 *
 * If this is not the case or you just want to try ILIAS, you'll find
 * us at:
 * https://www.ilias.de
 * https://github.com/ILIAS-eLearning
 *
 *********************************************************************/
```

Your license header:

```
/******************************************************************************
 *
 * This file is part of ILIAS, a powerful learning management system.
 *
 * ILIAS is licensed with the GPL-3.0, you should have received a copy
 * of said license along with the source code.
 *
 * If this is not the case or you just want to try ILIAS, you'll find
 * us at:
 * https://www.ilias.de
 * https://github.com/ILIAS-eLearning
 *
 *****************************************************************************/
```

Not sure where the difference comes from, or do I miss something? See https://docu.ilias.de/goto_docu_wiki_wpage_7116_1357.html

**Comment by alex40724** [comment]
@chfsx  Thanks for adding all the types.

A second thing related to the license:

Why putting the license under the "use" statements. I am never interested in the license but often in the use statements, so I would prefer them to have "near" the actual code. License first thing after <?php tag and "use" statements follow.

**Comment by chfsx** [comment]
Hi @alex40724 
Thanks for the hint! I'm Using rector to add the License header automatically, there seem to be an outdated template in it. I'll fix that asap and update the PR. This is also where the header beneath the use statements comes from, I'll change that manually

File: Modules/Exercise/classes/BackgroundTasks/class.ilExDownloadSubmissionsZipInteraction.php [hunk 1]
```diff
@@ -28,6 +39,9 @@ public function __construct()
     }
 
 
+    /**
+     * @return \ILIAS\BackgroundTasks\Types\SingleType[]
+     */
```
Comment (alex40724) [inline Modules/Exercise/classes/BackgroundTasks/class.ilExDownloadSubmissionsZipInteraction.php:47]: @chfsx Great!

**Comment by alex40724** [comment]
@chfsx Thanks. All other changes look good to me. I will merge after the license fix and take care of the TODOs afterwards.

**Comment by chfsx** [comment]
@alex40724 License-Header should now be correct and at a better position when files have namespaces, includes or use statements

**Comment by chfsx** [comment]
Tests are ok now

**Comment by alex40724** [comment]
I fixed the open Todos.

**Pull request merged**